### PR TITLE
WIP: Return admins of a group first in the list of group users

### PR DESCRIPTION
**Pull request by lukebarnard1**
This introduces a ORDER BY that can be optionally added to _simple_select_list transactions'

### CODE REVIEW

**Comment by dbkr** [comment]
(apart from the broken tests)

**Comment by lukebarnard1** [comment]
The tests are broken because postgres doesn't like it when you `ORDER BY` a string constant... SQLite seems more than happy to do a weird thing.

File: synapse/storage/_base.py [hunk 1]
```diff
@@ -650,20 +653,29 @@ def _simple_select_list_txn(cls, txn, table, keyvalues, retcols):
                 column names and values to select the rows with, or None to not
                 apply a WHERE clause.
             retcols (iterable[str]): the names of the columns to return
+            order (str): order the select by this column
+            order_asc (bool): order ascending (if order specified). Default True.
```
Comment (erikjohnston) [inline synapse/storage/_base.py]: TBH I would probably just have an `order` param that could be anything, e.g. `"foo ASC, bar DESC"`

File: synapse/storage/_base.py [hunk 2]
```diff
@@ -650,20 +653,29 @@ def _simple_select_list_txn(cls, txn, table, keyvalues, retcols):
...
         """
+        query_values = []
         if keyvalues:
             sql = "SELECT %s FROM %s WHERE %s" % (
                 ", ".join(retcols),
                 table,
                 " AND ".join("%s = ?" % (k, ) for k in keyvalues)
             )
-            txn.execute(sql, keyvalues.values())
+            query_values.extend(keyvalues.values())
```
Comment (erikjohnston) [inline synapse/storage/_base.py]: For performance reasons, I'd probably rely on variable hoisting and do:

```python
if keyvalues:
    ...
    query_values = keyvalues.values()
else:
    query_values = []
```

**Comment by erikjohnston** [comment]
Is the intention here that we the groups users API guarantees this order? Or is this just an arbitrary ordering that synapse decides on?

**Comment by lukebarnard1** [comment]
I'd be eternally grateful if @erikjohnston were to help me out with this, I'm really not sure how to grok the test failures

**Comment by erikjohnston** [comment]
@matrixbot retest this please

File: synapse/storage/_base.py (hunk 2 above)
Comment (lukebarnard1) [reply synapse/storage/_base.py]: wfm!

**Comment by lukebarnard1** [comment]
Closing this as it's a bit of a hack and we'd get the same result with some client-side ordering on the entire list of users (which we get now anyway because pagination hasn't been implementing).

**Pull request closed without merging**